JavaScript/JavaScriptSDK.Tests/CheckinTests/Telemetry/PageViewPerformance.tests.ts

```diff
@@ -124,6 +150,61 @@ class PageViewPerformanceTelemetryTests extends ContractTestHelper {
             }
         });
 
+        this.testCase({
+            name: name + "PageViewPerformanceTelemetry checks if any duration exceeds 5 minutes",
+            test: () => {
+                // see comment PageViewPerformance constructor on how timing data is calculated
+                // here we set values, so each metric will be exactly 300000 (5min).
+                let timingModifiers = [(timing) => timing.loadEventEnd = 300001,
+                    (timing) => timing.connectEnd = 300001,
+                    (timing) => timing.responseStart = 300003,
+                    (timing) => timing.responseEnd = 300030,
+                    (timing) => timing.loadEventEnd = 300042];
+
+                for (var i = 0; i < timingModifiers.length; i++) {
+
+                    var timing = <PerformanceTiming>{};
+                    timing.navigationStart = 1;
+                    timing.connectEnd = 2;
+                    timing.requestStart = 3;
+                    timing.responseStart = 30;
+                    timing.responseEnd = 42;
+                    timing.loadEventEnd = 60;
+
+                    // change perf timing value
+                    timingModifiers[i](timing);
+
+                    var timingSpy = this.sandbox.stub(Microsoft.ApplicationInsights.Telemetry.PageViewPerformance, "getPerformanceTiming", () => {
+                        return timing;
+                    });
+
+                    var actualLoggedMessage: string = "";
+                    Microsoft.ApplicationInsights._InternalLogging.verboseLogging = () => true;
+                    var loggingSpy = this.sandbox.stub(Microsoft.ApplicationInsights._InternalLogging, "warnToConsole", (m) => actualLoggedMessage = m);
+
+                    var telemetry = new Microsoft.ApplicationInsights.Telemetry.PageViewPerformance("name", "url", 0);
+                    Assert.equal(false, telemetry.getIsValid());
+
+                    var data = telemetry;
+
+                    Assert.equal(undefined, data.perfTotal);
+                    Assert.equal(undefined, data.networkConnect);
+                    Assert.equal(undefined, data.sentRequest);
+                    Assert.equal(undefined, data.receivedResponse);
+                    Assert.equal(undefined, data.domProcessing);
+
+                    if (i === 0) {
+                        // check props only for the first timingModifier
+                        Assert.equal("AI (Internal): NONUSRACT_MaxDurationExceeded message:\"exceeded maximum duration value (5min). Browser perf data won't be sent.\" props:\"{total:300000,network:1,request:27,response:12,dom:299959}\"", actualLoggedMessage);
+                    } else {
+                        Assert.ok(actualLoggedMessage.lastIndexOf("AI (Internal): NONUSRACT_MaxDurationExceeded message:\"exceeded maximum duration value (5min)", 0) === 0);
```

Comment:
Durations reported by bots are >1h (usually in a 2-5h range). I can change the code to exclude durations longer than 1h. I can also check a user agent to see if it's a GoogleBot. 

I don't want to remove a valid page load duration (event if it's 5min+), so I'll change it to use more strict rules first (duration >1h, user agent == GoogleBog). If our users will still see invalid values reported by navigation.timing API, we will make rules less strict.